code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -92,62 +78,84 @@ func (zcn *ZCNSmartContract) AddAuthorizer(tran *transaction.Transaction, inputD
 		return "", err
 	}
 
-	// compare the global min of authorizerNode Authorizer to that of the transaction amount
-	if globalNode.MinStakeAmount > state.Balance(tran.Value*1e10) {
-		msg := fmt.Sprintf("min stake amount '(%d)' > transaction value '(%d)'", globalNode.MinStakeAmount, tran.Value)
-		err = common.NewError(code, msg)
-		Logger.Error("min stake amount > transaction value", zap.Error(err))
-		return "", err
-	}
-
-	// Create Authorizer instance
-
-	authorizer = NewAuthorizer(authorizerID, authorizerPublicKey, authorizerURL)
-
-	// Dig pool for authorizer
-
-	transfer, response, err := authorizer.Staking.DigPool(tran.Hash, tran)
-	if err != nil {
-		err = common.NewError(code, fmt.Sprintf("error digging pool, err: (%v)", err))
-		return "", err
-	}
-
-	err = ctx.AddTransfer(transfer)
-	if err != nil {
-		msg := "Error: '%v', transaction.ClientId: '%s', transaction.ToClientId: '%s', transfer.ClientID: '%s', transfer.ToClientID: '%s'"
-		err = common.NewError(
-			code,
-			fmt.Sprintf(
-				msg,
-				err,
-				authorizerID,
-				recipientID,
-				transfer.ClientID,
-				transfer.ToClientID,
-			),
-		)
-		return "", err
-	}
+	// Check existing Authorizer
 
-	err = authorizer.Save(ctx)
-	if err != nil {
-		msg := fmt.Sprintf("error saving authorizer(authorizerID: %v), err: %v", authorizerID, err)
+	authorizer, err = GetAuthorizerNode(authorizerID, ctx)
+	if err == nil && authorizer != nil {
+		msg := fmt.Sprintf("authorizer(authorizerID: %v) already exists: %v", authorizerID, err)
 		err = common.NewError(code, msg)
-		Logger.Error("saving authorizer node", zap.Error(err))
-		return "", err
-	}
+		Logger.Warn("get authorizer node", zap.Error(err))
 
-	ev, err := authorizer.ToEvent()
-	if err != nil {
-		msg := fmt.Sprintf("error marshalling authorizer(authorizerID: %v) to event, err: %v", authorizerID, err)
-		err = common.NewError(code, msg)
-		Logger.Error("emitting event", zap.Error(err))
-		return "", err
+		// create stake pool for the validator to count its rewards
+		//var sp *stakePool
+		//sp, err = zcn.getOrUpdateStakePool(globalNode, authorizerID, spenum.Authorizer, params.StakePoolSettings, ctx)
+		//if err != nil {
+		//	return "", common.NewError(code, "get or create stake pool error: "+err.Error())
+		//}
+		//if err = sp.save(zcn.ID, authorizerID, ctx); err != nil {
+		//	return "", common.NewError(code, "saving stake pool error: "+err.Error())
+		//}
+	} else {
+
+		// compare the global min of authorizerNode Authorizer to that of the transaction amount
+		if globalNode.MinStakeAmount > state.Balance(tran.Value*1e10) {
+			msg := fmt.Sprintf("min stake amount '(%d)' > transaction value '(%d)'",
+				globalNode.MinStakeAmount, tran.Value)
+			err = common.NewError(code, msg)
+			Logger.Error("min stake amount > transaction value", zap.Error(err))
```

Comment:
most likely we will log error higher in the stack, so this can be removed `zap.Error(err)`

Reply:
No, this is the last call to log everything until the call is passed to transaction caller